Description: Avoid requesting remote SSL certificate before SNI.
Origin: commit, revision id: jelmer@jelmer.uk-20170111224745-xs2hj04l6xhswuse
Author: Jelmer VernooÄ³ <jelmer@jelmer.uk>
Bug: https://launchpad.net/bugs/1089352
Last-Update: 2017-01-11
Applied-Upstream: no
X-Bzr-Revision-Id: jelmer@jelmer.uk-20170111224745-xs2hj04l6xhswuse

=== modified file 'a/breezy/transport/http/_urllib2_wrappers.py'
Index: unstable/breezy/transport/http/_urllib2_wrappers.py
===================================================================
--- unstable.orig/breezy/transport/http/_urllib2_wrappers.py
+++ unstable/breezy/transport/http/_urllib2_wrappers.py
@@ -78,14 +78,9 @@ from ...sixish import (
 )
 
 try:
-    _ = (ssl.match_hostname, ssl.CertificateError)
+    ssl.CertificateError
 except AttributeError:
     # Provide fallbacks for python < 2.7.9
-    def match_hostname(cert, host):
-        trace.warning(
-            '%s cannot be verified, https certificates verification is only'
-            ' available for python versions >= 2.7.9' % (host,))
-    ssl.match_hostname = match_hostname
     ssl.CertificateError = ValueError
 
 
@@ -457,7 +452,7 @@ class HTTPSConnection(AbstractHTTPConnec
         try:
             ssl_sock = ssl.wrap_socket(
                 self.sock, self.key_file, self.cert_file,
-                cert_reqs=cert_reqs, ca_certs=ca_certs)
+                cert_reqs=cert_reqs, ca_certs=ca_certs, server_hostname=host)
         except ssl.SSLError:
             trace.note(
                 "\n"
@@ -466,9 +461,6 @@ class HTTPSConnection(AbstractHTTPConnec
                 "Pass -Ossl.cert_reqs=none to disable certificate "
                 "verification entirely.\n")
             raise
-        if cert_reqs == ssl.CERT_REQUIRED:
-            peer_cert = ssl_sock.getpeercert()
-            ssl.match_hostname(peer_cert, host)
 
         # Wrap the ssl socket before anybody use it
         self._wrap_socket_for_reporting(ssl_sock)
